components/availability/src/Availability.svelte

```diff
@@ -824,58 +848,94 @@
     }
   })();
 
+  // If 2 slots bump up to each other consider them a single group of time
+  function groupConsecutiveTimeslots(slots: PreDatedTimeSlot[] = []) {
+    return slots.reduce((groups, slot) => {
+      const prevSlot: PreDatedTimeSlot = groups[groups.length - 1];
+      if (prevSlot && prevSlot.end_time === slot.start_time) {
+        prevSlot.end_time = slot.end_time;
+      } else {
+        groups.push({ ...slot }); // TODO: types ¯\_(ツ)_/¯
+      }
+      return groups;
+    }, []);
+  }
+
   async function getAvailability(forceReload = false) {
     loading = true;
     let freeBusyCalendars: any = [];
     // Free-Busy endpoint returns busy timeslots for given email_ids between start_time & end_time
-    let consolidatedAvailabilityForGivenDay: AvailabilityResponse[] = [];
-
-    if (Array.isArray(email_ids) && email_ids.length > 0) {
-      consolidatedAvailabilityForGivenDay =
-        consolidatedAvailabilityForGivenDay.concat(
-          await $AvailabilityStore[
-            JSON.stringify({ ...getAvailabilityQuery(), forceReload })
-          ],
-        );
-    }
+
+    type fetchableCalendarUser = { email: string; token?: string };
+    let calendarsToFetch: fetchableCalendarUser[] = email_ids.map((email) => {
+      return {
+        email,
+      };
+    });
+
+    // If the booking user and access token are passed in, fetch their calendars as well.
+    // TODO: dont include them in the main list, as they shouldn't contribute to partial slot availability.
     if (booking_user_email && booking_user_token) {
-      consolidatedAvailabilityForGivenDay =
-        consolidatedAvailabilityForGivenDay.concat(
-          await $AvailabilityStore[
+      calendarsToFetch.push({
+        email: booking_user_email,
+        token: booking_user_token,
+      });
+      required_participants.push(booking_user_email);
```

Comment:
note that this will currently mutate the array passed in as a prop to the component - seems like unexpected behaviour for the user

Reply:
Thanks; made this an internal camelCased property and pushed to it. (also de-duped it in doing so)